api/agentic/v1alpha1/accesspolicy_types.go

```diff
@@ -0,0 +1,179 @@
+// the rule matches a request if either:
+// - the request's identity is in `Identities`
+// - OR the request's Serviceaccount matches an entry in `ServiceAccounts`.
+//
+// Each list within the fields (e.g. `Identities`) is itself an OR list.
+//
+// If this struct is omitted in a rule, it matches any source.
+//
+// <gateway:util:excludeFromCRD> NOTE: In the future, if there’s a need to express more complex
+// logical conditions (e.g. requiring a request to match multiple
+// criteria simultaneously—logical AND), we may evolve this API
+// to support richer match expressions or logical operators. </gateway:util:excludeFromCRD>
+type Source struct {
+
+	// Identities specifies a list of identities that are matched by this rule.
+	// A request's identity must be present in this list to match the rule.
+	//
+	// Identities must be specified as SPIFFE-formatted URIs following the pattern:
+	//   spiffe://<trust_domain>/<workload-identifier>
+	//
+	// While the exact workload identifier structure is implementation-specific,
+	// implementations are encouraged to follow the convention of
+	// `spiffe://<trust_domain>/ns/<namespace>/sa/<serviceaccount>`
+	// when representing Kubernetes workload identities.
+	//
+	// +optional
+	Identities []string `json:"identities,omitempty"`
+
+	// ServiceAccounts specifies a list of Kubernetes Service Accounts that are
+	// matched by this rule. A request originating from a pod associated with
+	// one of these Serviceaccounts will match the rule.
+	//
+	// Values must be in one of the following formats:
+	//   - "<namespace>/<serviceaccount-name>": A specific Serviceaccount in a namespace.
+	//   - "<namespace>/*": All Serviceaccounts in the given namespace.
+	//   - "<serviceaccount-name>": a Serviceaccount in the same namespace as the policy.
+	//
+	// Use of "*" alone (i.e., all Serviceaccounts in all namespaces) is not allowed.
+	// To select all Serviceaccounts in the current namespace, use "<namespace>/*" explicitly.
+	//
+	// Example:
+	//   - "default/bookstore" → Matches Serviceaccount "bookstore" in namespace "default"
+	//   - "payments/*" → Matches any Serviceaccount in namespace "payments"
+	//   - "frontend" → Matches "frontend" Serviceaccount in the same namespace as the policy
+	//
+	// The ServiceAccounts listed here are expected to exist within the same
+	// trust domain as the targeted workload, which in many environments means
+	// the same Kubernetes cluster. Cross-cluster or cross-trust-domain access
+	// should instead be expressed using the `Identities` field.
+	//
+	// +optional
+	ServiceAccounts []string `json:"serviceAccounts,omitempty"`
+}
+
+// AccessPolicyStatus defines the observed state of AccessPolicy.
+type AccessPolicyStatus struct {
+	// For Policy Status API conventions, see:
+	// https://gateway-api.sigs.k8s.io/geps/gep-713/#the-status-stanza-of-policy-objects
+	//
+	// Ancestors is a list of ancestor resources (usually Backend) that are
```

Comment:
If this is cut and paste, consider just having the link -- it's going to be hard to maintain the text here if the definition changes.

Reply:
SG. Done